src/category_theory/lifting_properties.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Scholbach
+-/
+import category_theory.category
+import category_theory.arrow
+import category_theory.functor
+import category_theory.limits.preserves.shapes.terminal
+
+/-!
+# Lifting properties
+
+This file defines the lifting property of two arrows in a category and show basic properties of 
+this notion. 
+We also construct the subcategory consisting of those morphisms which have the right lifting 
+property with respect to arrows in a given diagram.
+
+## Main results
+- `has_lifting_property`: the definition of the lifting property
+- `iso_has_right_lifting_property`: any isomorphism satisfies the right lifting property (rlp)
+- `id_has_right_lifting_property`: any identity has the rlp
+- `right_lifting_property_initial_iff`: spells out the rlp against a map whose source is an 
+initial object
+- `right_lifting_subcat`: given a functor F : D ⥤ arrow C, we construct the subcategory of those 
+morphisms p in C that satisfy the rlp w.r.t. F(i), for any object i in D.
```

Comment:
```suggestion
- `right_lifting_subcat`: given a functor `F : D ⥤ arrow C`, we construct the subcategory of those 
morphisms `p` in `C` that satisfy the rlp w.r.t. `F.obj i`, for any object `i` in `D`.
```
(In comments, use either backticks or dollars to delineate maths, preferring backticks and Lean syntax when plausible.)

Reply:
Done (except for the little commutative squares).